src/parser/SGTypes.ts

```diff
@@ -861,25 +859,32 @@ export interface SGReferences {
 
 export class SGAst {
 
-    constructor(
-        public prologElement?: SGProlog,
-        public rootElement?: SGElement,
-        public componentElement?: SGComponent
-    ) {
+    constructor(options: {
+        prolog?: SGProlog;
+        root?: SGElement;
+        component?: SGComponent;
+    } = {}) {
+        this.prolog = options.prolog;
+        this.root = options.root;
+        this.component = options.component;
     }
 
+    public readonly prolog?: SGProlog;
+    public readonly root?: SGElement;
+    public readonly component?: SGComponent;
```

Comment:
We actually intentially chose the names `*Element` in #818 to align with the way the HTML dom references their items. Let's keep these original names.